src/jarabe/main.py

```diff
@@ -347,6 +347,43 @@ def setup_fonts():
     settings.set_property("gtk-font-name", "%s %f" % (face, size))
 
 
+def setup_proxy():
+    g_mode = Gio.Settings('org.gnome.system.proxy').get_string('mode')
+    if g_mode == 'manual':
+        g_http = Gio.Settings('org.gnome.system.proxy.http')
+        g_http_host = g_http.get_string('host')
+        g_http_port = str(g_http.get_int('port'))
+        if(g_http.get_boolean('use-authentication')):
+            text_to_set = "http://" + g_http.get_string('authentication-user') + \
```

Comment:
This is a very complex assignment to text_to_set, because you use string concatenation operator "+".  Makes security review more difficult.  Simplify using string format operators, like .format or "%".